parity/cli/mod.rs

```diff
@@ -17,387 +17,949 @@
+				"Import wallet",
+
+				ARG arg_wallet_import_password: (Option<String>) = None,
+				"--password=[FILE]",
+				"Path to the password file",
+
+				ARG arg_wallet_import_path: (Option<String>) = None,
+				"<PATH>",
+				"Path to the wallet",
+			}
+		}
+
+		CMD cmd_import
+		{
+			"Import blockchain",
+
+			ARG arg_import_format: (Option<String>) = None,
+			"--format=[FORMAT]",
+			"Import in a given format. FORMAT must be either 'hex' or 'binary'. (default: auto)",
+
+			ARG arg_import_file: (Option<String>) = None,
+			"[FILE]",
+			"Path to the file to import from",
+		}
+
+		CMD cmd_export
+		{
+			"Export blockchain",
+
+			CMD cmd_export_blocks
+			{
+				"Export blocks",
+
+				ARG arg_export_blocks_format: (Option<String>) = None,
+				"--format=[FORMAT]",
+				"Export in a given format. FORMAT must be either 'hex' or 'binary'. (default: binary)",
+
+				ARG arg_export_blocks_from: (String) = "1",
+				"--from=[BLOCK]",
+				"Export from block BLOCK, which may be an index or hash.",
+
+				ARG arg_export_blocks_to: (String) = "latest",
+				"--to=[BLOCK]",
+				"Export to (including) block BLOCK, which may be an index, hash or latest.",
+
+				ARG arg_export_blocks_file: (Option<String>) = None,
+				"[FILE]",
+				"Path to the exported file",
+			}
+
+			CMD cmd_export_state
+			{
+				"Export state",
+
+				FLAG flag_export_state_no_storage: (bool) = false,
+				"--no-storage",
+				"Don't export account storage.",
+
+				FLAG flag_export_state_no_code: (bool) = false,
+				"--no-code",
+				"Don't export account code.",
+
+				ARG arg_export_state_min_balance: (Option<String>) = None,
+				"--min-balance=[WEI]",
+				"Don't export accounts with balance less than specified.",
+
+				ARG arg_export_state_max_balance: (Option<String>) = None,
+				"--max-balance=[WEI]",
+				"Don't export accounts with balance greater than specified.",
+
+				ARG arg_export_state_at: (String) = "latest",
+				"--at=[BLOCK]",
+				"Take a snapshot at the given block, which may be an index, hash, or latest. Note that taking snapshots at non-recent blocks will only work with --pruning archive",
+
+				ARG arg_export_state_format: (Option<String>) = None,
+				"--format=[FORMAT]",
+				"Export in a given format. FORMAT must be either 'hex' or 'binary'. (default: binary)",
+
+				ARG arg_export_state_file: (Option<String>) = None,
+				"[FILE]",
+				"Path to the exported file",
+			}
+		}
+
+		CMD cmd_signer
+		{
+			"Manage signer",
+
+			CMD cmd_signer_new_token {
+				"Generate new token",
+			}
+
+			CMD cmd_signer_list {
+				"List",
+			}
+
+			CMD cmd_signer_sign
+			{
+				"Sign",
+
+				ARG arg_signer_sign_password: (Option<String>) = None,
+				"--password=[FILE]",
+				"Path to the password file",
+
+				ARG arg_signer_sign_id: (Option<usize>) = None,
+				"[ID]",
+				"ID",
+			}
+
+			CMD cmd_signer_reject
+			{
+				"Reject",
+
+				ARG arg_signer_reject_id: (Option<usize>) = None,
+				"<ID>",
+				"ID",
+			}
+		}
+
+		CMD cmd_snapshot
```

Comment:
Should those two be grouped together?

Reply:
what do you mean?

Reply:
like:
`parity snapshot take` and `parity snapshot restore`

Right now while `parity snapshot` is self-describing, I had to check the docs what `parity restore` is supposed to do.

Reply:
yeah, makes sense! I too think it'd be better this way, but then the usage would change from the current version (currently it's `parity snapshot` and `parity restore`; I didn't change the name of the subcommands or options, simply refactored). It wouldn't be hard to find the new subcommand, but it'd be a breaking change in case people have scripts running with the current syntax for example. maybe we can keep this PR a refactoring one?

Reply:
A separate PR is fine for me, although we already break backward compatibility by disallowing options after the command name.